Woof/Woof/NetworkLayer/Endpoint/Encodable+asDictionary.swift

```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+extension Encodable {
+    /// Converts the object to a dictionary representation.
+    ///
+    /// - Throws: An error of type `Error` if encoding or serialization fails.
+    ///
+    /// - Returns: A dictionary representation of the encoded object.
+    func asDictionary() throws -> [String: Any] {
+        let encoder = JSONEncoder()
+        encoder.keyEncodingStrategy = .convertToSnakeCase
+
+        let data = try encoder.encode(self)
+        let dictionary = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: Any]
```

Comment:
Why do we need this option here?
```suggestion
        let dictionary = try JSONSerialization.jsonObject(with: data) as? [String: Any]
```

Reply:
Fixed

Reply:
I don't see you have fixed it.

Reply:
My bad, stashed changes and forgot to apply them back.